pkg/route/api/validation/validation.go

```diff
@@ -26,7 +26,8 @@ func ValidateRoute(route *routeapi.Route) field.ErrorList {
 
 	//host is not required but if it is set ensure it meets DNS requirements
 	if len(route.Spec.Host) > 0 {
-		if len(kvalidation.IsDNS1123Subdomain(route.Spec.Host)) != 0 {
+		hostname, _ := routeapi.NormalizeWildcardHost(route.Spec.Host)
+		if len(kvalidation.IsDNS1123Subdomain(hostname)) != 0 {
```

Comment:
It would have failed on a 1.2/1.3 cluster as well - wouldn't be a valid host name.  In any case, the router doesn't allow wildcards by default - you have to enable it and if so, you could whitelist/blacklist the names you allow.
I can add a check to the admission side to ensure that 1 liner - host name is valid but it does add some extra processing time per route.